Fixes three things Hangprinter needs

tobbelobb opened the pull request:
... because otherwise, the effect of the callback is canceled by an
assignment in the main update() loop, that calculates input_pos
based on steps_, if we're in step/dir mode.

- Fixes a steps_ underflow/wrapping bug

tobbelobb committed:
Sets steps_ in CANSimple::set_input_pos_callback

... because otherwise, the effect of the callback is canceled by an
assignment in the main update() loop, that calculates input_pos
based on steps_, if we're in step/dir mode.

- Fixes a steps_ underflow/wrapping bug

tobbelobb committed:
Puts fmodf_pos back into the steps_ assignment

tobbelobb committed:
Stands still upon switching to position mode via can

tobbelobb committed:
Making circular_setpoints optional in step/dir mode

.. because circular_setpoints was broken beyond repair.

madcowswe commented on Firmware/communication/can/can_simple.cpp:
```diff
@@ -269,7 +269,20 @@ void CANSimple::set_input_torque_callback(Axis& axis, const can_Message_t& msg)
+        }
+
+        axis.controller_.set_input_pos_and_steps(estimate);
+    }
```
This reaction on writing the control mode is appropriate, but we need the behavior to be consistent when set from other sources than CAN. My suggestion is:
1. Merge to a reusable function this snippet with the "// To avoid any transient on startup" section in `axis.cpp start_closed_loop_control`. You could call it `set_setpoint_to_estimate`.
2. Add a `c_setter` tag to the `control_mode` endpoint in `odrive-interface.yaml`, so this can get executed when we write it on USB or UART.
3. Add an entry in the changelog about this reaction on control mode change.

madcowswe reviewed (changes requested):
Looks good in general, thank you. We just need to make the behavior consistent on the other (non-CAN) interfaces. See my other note.

tobbelobb committed:
Makes behavior consistent across interfaces

Defines and uses set_setpoint_to_estimate() reusable function
Uses it in three places:
 - axis.cpp to avoid transient on startup
 - can_simple.cpp to avoid sudden movement upon mode change
 - odrive-interface.yaml just to make it executable via USB and UART

Swaps set_input_pos() to set_input_pos_and_steps() so that we get the
same behavior when changing input_pos over USB and UART as we get when
we change it via CAN.
